src/main/java/org/apache/brooklyn/cloudfoundry/location/CloudFoundryPaasClient.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.brooklyn.cloudfoundry.location;
+
+import java.io.IOException;
+import java.util.Map;
+
+import org.cloudfoundry.client.lib.CloudCredentials;
+import org.cloudfoundry.client.lib.CloudFoundryClient;
+
+public class CloudFoundryPaasClient {
+
+    private final CloudFoundryPaasLocation location;
+    private CloudFoundryClient client;
+
+    public CloudFoundryPaasClient(CloudFoundryPaasLocation location) {
+        this.location = location;
+        setUpClient();
+    }
+
+    private void setUpClient() {
+        if (client == null) {
+            CloudCredentials credentials =
+                    new CloudCredentials(
+                            location.getConfig(CloudFoundryPaasLocationConfig.ACCESS_IDENTITY),
+                            location.getConfig(CloudFoundryPaasLocationConfig.ACCESS_CREDENTIAL));
+            client = new CloudFoundryClient(credentials, null);
+            client.login();
+        }
+    }
+
+    public String deploy(Map<?, ?> params) {
+        String domain;
+        //TODO using client to deploy application
+        //client.createApplication(applicationName, buildpack, memory, disk,...)
+        //pushArtifact(applicationName, artifact);
+
+        //returns the domain of the deployed application
+        return null;
+    }
+
+    public void pushArtifact(String applicationName, String artifact) {
```

Comment:
You can create an application  in a Cloud Foundry platform. This application contains the domain, the host, the application name the resource profile, the buildpack and so on, but does not the artifact. After, once application exists, you can upload an artifact to the application. So, CloudFoundry API provides these different services. So, the `CloudFoundryPaasLocation` provides these two different operations. The entity has to deploy the application and push the artifact. In any case, we could do the following, `CFPaasLocation` can provides three different operations:
- createApplication
- pushArtifact
- deployApplication (which will call to create the application and push the artifact methods). 

Then, the entity could use `cfPaasLocation.deploy(...)`  to deploy the artifact.
If we provides a method to push the artifact we will be able to provides a effector wich allows to update an application in order to deploy a newer application artifact.